classes/Product.php

```diff
@@ -958,7 +959,7 @@ public function toggleStatus()
         if ($this->active) {
             //case where active will be false after parent::toggleStatus()
             $this->id_type_redirected = 0;
-            $this->redirect_type = '404';
+            $this->redirect_type = ProductInterface::REDIRECT_CATEGORY_301;
```

Comment:
you'd better use https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Response.php#L36

Reply:
No, it's specific to product redirect even if the code is the same, we need to keep it in its own interface.

Reply:
yep, you have two 301 redirections, 301-category and 301-product so we can't simply use a 301 integer